server/src/lib/basket.ts

```diff
@@ -11,80 +11,154 @@ export const API_URL =
     ? 'https://basket.mozilla.org'
     : 'https://basket-dev.allizom.org';
 
+const basketConfig = {
+  uri: API_URL + '/news/common-voice-goals/',
+  method: 'POST',
+  headers: {
+    'x-api-key': BASKET_API_KEY,
+  },
+};
+
 function toISO(date: string) {
   return date ? new Date(date).toISOString().slice(0, -5) + 'Z' : null;
 }
 
-export async function sync(client_id: string) {
-  await computeGoals(client_id);
-
+async function getCurrentStatus(client_id: string) {
   const [[row]] = await db.query(
+    `SELECT u.*,
+      NOW() AS new_last_active,
+      HOUR(TIMEDIFF(NOW(), u.last_active)) AS hours_elapsed
+    FROM user_client_newsletter_prefs AS u
+    WHERE client_id = ?`,
+    [client_id]
+  );
+
+  return row;
+}
+
+function updateLastActive(currentUserStats: any) {
+  db.query(
+    `UPDATE user_client_newsletter_prefs
+      SET last_active = ?
+    WHERE client_id = ?`,
+    [currentUserStats.new_last_active, currentUserStats.client_id]
+  );
+
+  const data = {
+    email: currentUserStats.email,
+    last_active_date: toISO(currentUserStats.new_last_active),
+  };
+
+  sendRequest({
+    ...basketConfig,
+    form: data,
+  }).catch(err => console.error(err.message));
+
+  console.log('basket', JSON.stringify(data, null, 2));
```

Comment:
Was this left in intentionally?

Reply:
Hmm originally yes, because there was a `console.log` in the original sync function and I just mirrored it. But now that I look at it, we don't want to be logging email addresses persistently, so I will take it out. TY!